File: contests/codeforces/2208/c/main.py
from functools import cache


def solve(N, c, p):
    ans = 0
    for i in range(N - 1, -1, -1):
        ans = max(ans, c[i] + (100 - p[i]) / 100 * ans)

    return ans


def main():
    t = int(input())
    for _ in range(t):
        n = int(input())

        C = []
        P = []
        for i in range(n):
            c, p = list(map(int, input().split()))
            C.append(c)
            P.append(p)

        print(solve(n, C, P))


if __name__ == "__main__":
    main()

File: contests/codeforces/2208/c/main.cpp
#include <bits/stdc++.h>
using namespace std;

#ifdef LOCAL
#include "algo/debug.h"
#define open(x) freopen(x, "r", stdin);
#else
#define open(x)
#endif

int N;
vector<int> C, P;

void solve() {
    cin >> N;
    C = vector<int>(N);
    P = vector<int>(N);

    for (int i=0; i<N; i++) {
        int c, p; cin >> c >> p;
        C[i] = c;
        P[i] = p;
    }

    long double ans = 0.0;
    for (int i=N-1; i>=0; i--) 
        ans = max(ans, C[i]*1.0L + ((100 - P[i])*1.0L/100) * ans);
    cout << ans << endl;
}

int main() {
    ios::sync_with_stdio(0); cin.tie(0);
    // open("input.txt");

    int t; cin >> t;
    while (t--) {
        solve();
    }
}
